### ISSUE TITLE
Echo bot example

### ISSUE THREAD

**Issue by mlite** [cee-studio/libdiscord#3]
Just like hello world for every person who starts programming,  we need a very simple echo bot example that just replies "hello world!" when a user type a command (you can pick any prefix) in a channel.    This needs to be so easy to get it up and running.

**Comment by mlite** [comment]
how reliable is the curl-web-socket implementation?

**Comment by LucasMull** [comment]
> how reliable is the curl-web-socket implementation?  I liked its simplicity. 

From my experience it's very easy to reason with. It's very simple, as you said, and doesn't add any extra dependencies which is a big plus.
From what I've gathered other libs like libwebsockets may sometimes conflict with libcurl, though I haven't tried it to know for sure.
If you want to check it out from the source: https://github.com/barbieri/barbieri-playground

The creator is also a pretty nice guy and answer his e-mails.

**Comment by mlite** [comment]
> > how reliable is the curl-web-socket implementation?  I liked its simplicity.
> 
> From my experience it's very easy to reason with. It's very simple, as you said, and doesn't add any extra dependencies which is a big plus and
> from what I've gathered other libs like libwebsockets may sometimes conflict with libcurl, though I haven't tried it to know for sure.
> If you want to check it out from the source: https://github.com/barbieri/barbieri-playground
> 
> The creator is also a pretty nice guy and answers his e-mails.

Yes, I filed an issue to ask him about this question.  Do we need to think about reconnecting or restart logic in case the ws connection is closed?

**Comment by LucasMull** [comment]
> Yes, I filed an issue to ask him about this question.  Do we need to think about reconnecting or restart logic in case the ws connection is closed? 
> 
> 
Yeah, this explains how we should implement it: https://discord.com/developers/docs/topics/gateway#resuming
